src/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosure.java

```diff
@@ -502,7 +502,7 @@ private void findBackupFile(HttpMessage originalMessage) throws Exception {
 				//but for a "Copy of" file, actually gives a 404 (for some unknown reason). We need to handle this case.
 				if ( 	( gives404s && requestStatusCode != HttpStatus.SC_NOT_FOUND) || 
 						(	(!gives404s) &&
-							requestStatusCode != HttpStatus.SC_NOT_FOUND &&
+								getBaseMsg().getResponseHeader().getStatusCode() == requestStatusCode &&
```

Comment:
Sorry, this looked good but after testing locally there's still a FP. Instead of `getBaseMsg()` we should use `nonexistfilemsg` (and check they are different), that's more reliable in case the original file does not exist (as in the reported case)?

We should also apply this fix to the test of the parent dir.